api/src/services/restaurants/restaurants.js

```diff
@@ -18,6 +18,16 @@ export const createRestaurant = ({ name, brandColor }) => {
           id: context.currentUser.id,
         },
       },
+      address: {
+        connectOrCreate: {
```

Comment:
We don't actually need `connectOrCreate` here, we should just be able to use `create` (see: https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#nested-writes).